tests/unit/job/test_load_config.py

```diff
@@ -413,6 +413,45 @@ def test_ignore_unknown_values_setter(self):
         config.ignore_unknown_values = True
         self.assertTrue(config._properties["load"]["ignoreUnknownValues"])
 
+    def test_json_extension_missing(self):
+        config = self._get_target_class()()
+        self.assertIsNone(config.json_extension)
+
+    def test_json_extension_hit(self):
```

Comment:
What is the intended difference between `test_*_hit` and `test_*_setter`

They go through very similar machinations.

Reply:
From my understanding, they are testing different behaviors, although I was following the format of the other tests, test_*_hit checks if the property is properly initialized or retrieved and test_*_set verifies that the setter correctly updates the internal state and reflects the new value, so its more testing the behavior of setting the property rather than testing the intial or retrieved value.

Reply:
I asked a little bit about this issue and one of the responses was pretty interesting and describes an interesting philosophy (i paraphrased it a bit and formatted it):

"Incidentally, ... I see it omitted in tests all the time: there should be an 

`assertFalse(config.key1)` then `config._properties["key1"] = True` and then the `assertTrue(config.key1)`

to make sure it actually registers a change in state."